### DISCUSSION TITLE
if add composite primary/unique key index, migration missing one foreign key index

### DISCUSSION THREAD

**Discussion by paul38467** [laravel/framework#34847] [Q&A] [answered]
- Laravel Version: 7.28.3
- Homestead 10.0.0

### Description:
if add composite primary/unique key index and with two foreign key index
in phpMyAdmin / HeidiSQL show missing one of the foreign key index

### Steps To Reproduce:
pivot table migration:

    public function up()
    {
        Schema::create('car_region', function (Blueprint $table) {
            $table->unsignedBigInteger('car_id');
            $table->unsignedSmallInteger('region_id');
            $table->timestamps();
    
            $table->primary(['car_id', 'region_id']);
            $table->foreign('car_id')->references('id')->on('cars');
            $table->foreign('region_id')->references('id')->on('regions');
        });
    }

### The index showing in phpMyAdmin / HeidiSQL is:

    Keyname                         Type    Unique  Packed  Column  
    _________________________________________________________________
    PRIMARY                         BTREE   Yes     No      car_id
                                                            region_id
    -----------------------------------------------------------------
    car_region_region_id_foreign    BTREE   No      No      region_id

### The expected index should be:

    Keyname                         Type    Unique  Packed  Column  
    _________________________________________________________________
    PRIMARY                         BTREE   Yes     No      car_id
                                                            region_id
    -----------------------------------------------------------------
    car_region_car_id_foreign       BTREE   No      No      car_id
    -----------------------------------------------------------------
    car_region_region_id_foreign    BTREE   No      No      region_id

### If remove the composite primary/unique key index, it show the correct both index:

            //$table->primary(['car_id', 'region_id']);
            $table->foreign('car_id')->references('id')->on('cars');
            $table->foreign('region_id')->references('id')->on('regions');

    Keyname                         Type    Unique  Packed  Column  
    _________________________________________________________________
    car_region_car_id_foreign       BTREE   No      No      car_id
    -----------------------------------------------------------------
    car_region_region_id_foreign    BTREE   No      No      region_id

**Comment by paulocuambe** [comment, marked as answer]
I think this is not supported check this comment in a related [issue](https://github.com/laravel/framework/issues/5355#issuecomment-152519370).

**Comment by parallels999** [comment]
https://github.com/topclaudy/compoships
